packages/commonwealth/server/util/roles.ts

```diff
@@ -124,7 +124,13 @@ export async function findAllCommunityRolesWithRoleAssignments(
   roleFindOptions.attributes = findOptions.attributes;
   roleFindOptions.order = findOptions.order;
 
-  const addresses = await models.Address.findAll(roleFindOptions);
+  let addresses;
+  try {
+    addresses = await models.Address.findAll(roleFindOptions);
```

Comment:
Why was this call throwing an error? @kurtisassad

Reply:
When were we seeing the `address_id: null`? Why not fix the issue directly rather than add a try/catch?

Reply:
@kurtisassad what is the root cause? To me, it looks like an issue with the tests not providing a mocked Address.id rather than with your code specifically. Why didn't we surface the underlying concern?

Reply:
@rbennettcw was there any reason for omitting address ids in the new test code?

Reply:
> Because the test is passing before my shim is in place, but failing after, this points to a failure in my shim, not the test itself I think.

This is not necessarily true. The test itself seems to make certain invalid assumptions about the required shape of the object, i.e. that omitting an Address id is safe, which "just so happened" to be safe before your changes, but was determined unsafe afterward.

It's good that you caught the case, insofar as we can ship more resilient code, but the root problem almost certainly is not that your code had a bug, but that your code made (valid) assumptions about the inputs that were not accounted for in designing the test. So, we should also fix the test as well.

Reply:
So for the shim, how should I handle these edge cases of passing undefined values for Address where clauses. Should I just remove the check for it altogether and change the test instead?

Reply:
> @rbennettcw was there any reason for omitting address ids in the new test code?

@jnaviask  The test was originally designed to include the minimal amount of mock data required to trigger the happy/unhappy path specific to the test.

Reply:
@kurtisassad update the current tests to provide address ids, so they continue to work as expected. Then write another test designed to fail, which we can handle with a try/catch.